pagopa/selfcare-ms-user-group
# [SELC-2434] add DELETED status for userGroup - refactor RequestParam for getAllGroups

**List of Changes**
1. refactor API deleteGroup - add DELETED status for userGroup
2. refactor API getUserGroups - changed RequestParameter status from UserGroupStatus to List<UserGroupStatus>

**Motivation and Context**
    Logical rather than physical deletion of UserGroups

**How Has This Been Tested?**
1. Try to delete userGroup by its ID
2. Check its status from getById
3. Try to call getUserGroups with some state's filter
